src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -106,11 +189,23 @@ public String toString() {
         return persons.asUnmodifiableObservableList();
     }
 
+    @Override
+    public ObservableList<Ingredient> getIngredientList() {
+        return inventory.getIngredientList();
+    }
+
+    @Override
+    public ObservableList<Recipe> getRecipeList() {
+        return cookbook.getRecipeList();
+    }
+
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof AddressBook // instanceof handles nulls
-                && persons.equals(((AddressBook) other).persons));
+                && persons.equals(((AddressBook) other).persons)
+                && inventory.equals(((AddressBook) other).inventory)
+                && cookbook.equals(((AddressBook) other).cookbook));
```

Comment:
should be cast to `Inventory` / `Cookbook`

Reply:
```AddressBook``` refers to the wrapper class for ```Inventory``` and ```Cookbook```, the ```equals()``` is in ```AddressBook```, so the comparison should be between ```AddressBook```s

Reply:
oops sorry